autogen/variables.tf

```diff
@@ -148,14 +148,9 @@ variable "disable_legacy_metadata_endpoints" {
 }
 
 variable "node_pools" {
-  type        = list(map(string))
-  description = "List of maps containing node pools"
-
-  default = [
-    {
-      name = "default-node-pool"
-    },
-  ]
+  type        = map
+  description = "map containing node pools"
```

Comment:
I don't think we need to make this into a map? I'd like to keep the external interface the same.

Reply:
Will keep the existing interface via 
```tf
node_pool_names = [for np in toset(var.node_pools) : np.name]
node_pools = zipmap(local.node_pool_names, tolist(toset(var.node_pools)))
```